react-components/hello-world.md

```diff
@@ -294,45 +283,63 @@ This file renders our first _JSX_ (described in another lesson).
 
 This code indicates that when we start our React webapp, it will _render_ an `App` component using `<App />`.
 
-Aside from seeing that this is how the `App` component gets loaded, we will rarely visit `src/index.js` in this curriculum, even though there are lines of code in this file that are worth exploring with bigger projects.
+Aside from seeing that this is how the `App` component gets loaded, we will rarely visit `src/main.jsx` in this curriculum, even though there are lines of code in this file that are worth exploring with bigger projects.
 
 Instead, we'll focus on working with the `App` component.
 
-### Exploring `App.js`
+### Exploring `App.jsx`
 
-Consider the contents of `src/App.js`.
+Consider the contents of `src/App.jsx`.
 
 <!-- prettier-ignore-start -->
 ```js
-import logo from './logo.svg';
-import './App.css';
+import { useState } from 'react'
+import reactLogo from './assets/react.svg'
+import viteLogo from '/vite.svg'
+import './App.css'
 
 function App() {
+  const [count, setCount] = useState(0)
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
+    <>
+      <div>
+        <a href="https://vitejs.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
+          <img src={reactLogo} className="logo react" alt="React logo" />
+        </a>
+      </div>
+      <h1>Vite + React</h1>
+      <div className="card">
+        <button onClick={() => setCount((count) => count + 1)}>
+          count is {count}
+        </button>
         <p>
-          Edit <code>src/App.js</code> and save to reload.
+          Edit <code>src/App.jsx</code> and save to test HMR
         </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+      </div>
+      <p className="read-the-docs">
+        Click on the Vite and React logos to learn more
+      </p>
+    </>
+  )
 }
 
-export default App;
+export default App
 ```
 <!-- prettier-ignore-end -->
 
-From reading `src/App.js`, answer the questions below. If a question involves an expression that hasn't been discussed yet, we can hypothesize what it might do based on what we know about other languages, or from what a natural language reading might imply.
+### !callout-warning
+
+## Don't Forget Your Semicolons!
+
+You might have noticed that the default template that Vite provides does not include semicolons. While there are a variety of styles when it comes to writing JavaScript and creating JSX objects, we encourage you to continue using semicolons!
```

Comment:
```suggestion
## Remember to Use Semicolons!

The default template that Vite provides does not include semicolons. While there are a variety of styles when it comes to writing JavaScript and creating JSX objects, we encourage using semicolons!
```

I'm so annoyed vite doesn't recommend semicolons, and this is annoying for students, but hey. We should make sure that any repos students will work from are updated to include the semicolon rule.

Here, we could add a paragraph like the following: (couldn't include in suggestion due to triple tick behavior)

We can add a line to the `.eslint.cjs` linter rule file to help remind us where they go! Within the `rules` section, add the following after any other rules specified by the template:
```js
semi: ['warn', 'always'],
```

I'd rather include an expandable example of what the rules section would look like. Something like:

detail header: Expand for an example of how the rules section could look after modification
expanded detail:
There could be different rules depending on the version of the `vite` tool used during the project setup. We added the `semi` rule to the end of the `rules` block.
```js
  rules: {
    'react-refresh/only-export-components': [
      'warn',
      { allowConstantExport: true },
    ],
    semi: ['warn', 'always'],
  },
```